src/json.hpp

```diff
@@ -1241,15 +1336,14 @@ class basic_json
 
     @sa @ref basic_json(const typename string_t::value_type*) -- create a
     string value from a character pointer
-    @sa @ref basic_json(const CompatibleStringType&) -- create a string value
+    @sa @ref basic_json(const CompatibleStringType&) -- create a string
+    value
     from a compatible string container
 
     @since version 1.0.0
     */
-    basic_json(const string_t& val)
-        : m_type(value_t::string), m_value(val)
-    {
-        assert_invariant();
+    basic_json(const string_t &val) : m_type(value_t::string), m_value(val) {
```

Comment:
My intuition says that we would like to use the trait system for constructing `basic_json` objects from all types that are not `basic_json`, since in case somebody has a different kind of representation in a string (e.g. XML), they could still define their own trait class to construct `basic_json` objects from that.